datalad/tests/test_tests_utils.py

```diff
@@ -223,38 +226,51 @@ def test_nok_startswith():
     assert_raises(AssertionError, nok_startswith, 'abc', 'abc')
 
 
-def test_assert_cwd_unchanged():
-    orig_dir = os.getcwd()
+def _test_assert_Xwd_unchanged(func):
+    orig_cwd = os.getcwd()
+    orig_pwd = getpwd()
 
     @assert_cwd_unchanged
     def do_chdir():
-        os.chdir(os.pardir)
+        func(os.pardir)
 
     assert_raises(AssertionError, do_chdir)
-    eq_(orig_dir, os.getcwd(),
-        "assert_cwd_unchanged didn't return us back to %s" % orig_dir)
+    eq_(orig_cwd, os.getcwd(),
+        "assert_cwd_unchanged didn't return us back to cwd %s" % orig_cwd)
+    eq_(orig_pwd, getpwd(),
+        "assert_cwd_unchanged didn't return us back to pwd %s" % orig_pwd)
 
+def test_assert_Xwd_unchanged():
+    yield _test_assert_Xwd_unchanged, os.chdir
```

Comment:
yes, useful at times ;)  but to be not abused ;) e.g. some projects (e.g. nibabel) now migrate to more scarce use of those (theoretically there could be a test yielding assert* statements) since complicates debugging